__tests__/app.test.js

```diff
@@ -1,54 +1,123 @@
-    test('returns status 200 and an object', () => {
-      return request(app).get('/api').then((response) => {
-        const endPoints = response.body.endPoint;
+          expect(topic.hasOwnProperty("slug"));
+          expect(topic.hasOwnProperty("description"));
+        });
         expect(response.status).toBe(200);
-        expect(typeof endPoints).toEqual('object');
       });
-    });
-    test('returns 200 and an object with the same structure as endpoint json file', () => {
-      return request(app).get('/api').then((response) => {
-        const endPoints = response.body.endPoint;
+  });
+});
+describe("GET/api", () => {
+  test("returns status 200 and an object", () => {
+    return request(app)
+      .get("/api")
+      .then((response) => {
+        const endPoints = response.body.endPoints;
+        expect(response.status).toBe(200);
+        expect(typeof endPoints).toEqual("object");
+      });
+  });
+  test("returns 200 and an object with the same structure and content as endpoint json file", () => {
+    return request(app)
+      .get("/api")
+      .then((response) => {
+        const endPoints = response.body.endPoints;
         expect(response.status).toBe(200);
         expect(endPoints).toEqual(endPointData);
       });
-    });
-  });
\ No newline at end of file
+  });
+});
+describe("GET/api/articles/:article_id", () => {
+  test("returns status 200 and an object", () => {
+    return request(app)
+      .get("/api/articles/1")
+      .then((response) => {
+        const article = response.body.article;
+        expect(response.status).toBe(200);
+        expect(typeof article).toEqual("object");
+      });
+  });
+  test("returns status 200 and an article object with correct properties", () => {
+    return request(app)
+      .get("/api/articles/1")
+      .then((response) => {
+        const article = response.body.article;
+        expect(response.status).toBe(200);
+        expect(article[0]).toEqual({
+          article_id: 1,
+          title: "Living in the shadow of a great man",
+          topic: "mitch",
+          author: "butter_bridge",
+          body: "I find this existence challenging",
+          created_at: "2020-07-09T20:11:00.000Z",
+          votes: 100,
+          article_img_url:
+            "https://images.pexels.com/photos/158651/news-newsletter-newspaper-information-158651.jpeg?w=700&h=700",
+        });
+      });
+  })
+  test("returns status 404 when id number is not in the database", () => {
+    return request(app)
+      .get("/api/articles/10090")
+      .then((response) => {
+        expect(response.status).toBe(404);
+        expect(response.body.msg).toBe("Not found");
+      });
+  });
+  test("returns status 404 when id number is not in the database", () => {
+    return request(app)
+      .get("/api/articles/banana")
+      .then((response) => {
+        expect(response.status).toBe(400);
+        expect(response.body.msg).toBe("Bad request");
+      });
+  });
```

Comment:
Good work implementing the error handling testing for an invalid id, and a valid id with no results. These are the correct status codes used.